Fix dfs server compilation on Windows (WAL stub + daemon mode)

## Context

The `dfs` server binary fails to compile on Windows (`GOOS=windows`) due to three issues:

### 1. Outdated WAL Windows stub (`pkg/cache/wal/mmap_windows.go`)

The Windows stub has fallen out of sync with the Unix implementation:
- `Recover()` returns `([]BlockWriteEntry, error)` but callers in `pkg/cache/cache.go:131` expect `(*RecoveryResult, error)`
- Missing `AppendBlockUploaded(payloadID string, chunkIdx, blockIdx uint32) error` method (called in `pkg/cache/flush.go:165`)

**Fix:** Update stub signatures to match current interface. All methods continue returning `ErrUnsupportedPlatform`.

### 2. Unix-only daemon mode (`cmd/dfs/commands/start.go`)

`startDaemon()` uses `syscall.SysProcAttr{Setsid: true}` which doesn't exist on Windows.

**Fix:** Extract into platform-specific files:
- `daemon_unix.go` (`//go:build !windows`) — current logic
- `daemon_windows.go` (`//go:build windows`) — returns error: "daemon mode not supported on Windows, use --foreground"

### 3. Unix-only stop command signals (`cmd/dfs/commands/stop.go`)

Uses `syscall.SIGKILL` which is unavailable on Windows.

**Fix:** Extract into platform-specific files:
- `stop_unix.go` — current SIGTERM/SIGKILL logic
- `stop_windows.go` — use `process.Kill()` for force stop, `os.Interrupt` for graceful

## Build errors

```
pkg/cache/cache.go:136:31: result.Entries undefined (type []wal.BlockWriteEntry has no field or method Entries)
pkg/cache/cache.go:154:25: result.UploadedBlocks undefined (type []wal.BlockWriteEntry has no field or method UploadedBlocks)
pkg/cache/flush.go:165:20: c.persister.AppendBlockUploaded undefined (type *wal.MmapPersister has no field or method AppendBlockUploaded)
```

## Acceptance criteria

- [ ] `go build ./cmd/dfs/` succeeds on Windows
- [ ] `dfs start --foreground` runs the server on Windows
- [ ] `dfs start` (daemon mode) returns a clear error on Windows
- [ ] `go vet ./...` passes on Windows

--- a/cmd/dfs/commands/daemon_unix.go
+++ b/cmd/dfs/commands/daemon_unix.go
@@ -1,0 +1,98 @@
+//go:build !windows
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+)
+
+// isProcessRunning reads a PID from the given file and checks whether
+// that process is still alive. Returns the PID and true if running,
+// or 0 and false otherwise.
+func isProcessRunning(pidPath string) (int, bool) {
+	pidData, err := os.ReadFile(pidPath)
+	if err != nil {
+		return 0, false
+	}
+
+	var pid int
+	if _, err := fmt.Sscanf(string(pidData), "%d", &pid); err != nil {
+		return 0, false
+	}
+
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return 0, false
+	}
+
+	if err := process.Signal(syscall.Signal(0)); err != nil {
+		return 0, false
+	}
+
+	return pid, true
+}
+
+// startDaemon starts the server as a background daemon process.
+func startDaemon() error {
+	stateDir := GetDefaultStateDir()
+	if err := os.MkdirAll(stateDir, 0755); err != nil {
+		return fmt.Errorf("failed to create state directory: %w", err)
+	}
+
+	pidPath := pidFile
+	if pidPath == "" {
+		pidPath = filepath.Join(stateDir, "dittofs.pid")
+	}
+
+	if pid, running := isProcessRunning(pidPath); running {
+		return fmt.Errorf("DittoFS is already running (PID %d)\nUse 'dfs stop' to stop the running instance", pid)
+	}
+	_ = os.Remove(pidPath)
+
+	logPath := logFile
+	if logPath == "" {
+		logPath = filepath.Join(stateDir, "dittofs.log")
+	}
+
+	executable, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("failed to get executable path: %w", err)
+	}
+
+	daemonArgs := []string{"start", "--foreground", "--pid-file", pidPath}
+	if GetConfigFile() != "" {
+		daemonArgs = append(daemonArgs, "--config", GetConfigFile())
+	}
+
+	cmd := exec.Command(executable, daemonArgs...)
+
+	logFileHandle, err := os.OpenFile(logPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open log file: %w", err)
+	}
+
+	cmd.Stdout = logFileHandle
+	cmd.Stderr = logFileHandle
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setsid: true,
+	}
+
+	if err := cmd.Start(); err != nil {
+		_ = logFileHandle.Close()
+		return fmt.Errorf("failed to start daemon: %w", err)
+	}
+
+	_ = logFileHandle.Close()
+
+	fmt.Printf("DittoFS started in background (PID %d)\n", cmd.Process.Pid)
+	fmt.Printf("  PID file: %s\n", pidPath)
+	fmt.Printf("  Log file: %s\n", logPath)
+	fmt.Println("\nUse 'dfs stop' to stop the server")
+	fmt.Println("Use 'dfs status' to check server status")
+
+	return nil
+}

--- a/cmd/dfs/commands/daemon_windows.go
+++ b/cmd/dfs/commands/daemon_windows.go
@@ -1,0 +1,11 @@
+//go:build windows
+
+package commands
+
+import "fmt"
+
+// startDaemon is not supported on Windows.
+// Use --foreground flag to run the server in the foreground.
+func startDaemon() error {
+	return fmt.Errorf("daemon mode is not supported on Windows, use --foreground")
+}

--- a/cmd/dfs/commands/start.go
+++ b/cmd/dfs/commands/start.go
@@ -4,9 +4,7 @@ import (
 	"context"
 	"fmt"
 	"os"
-	"os/exec"
 	"os/signal"
-	"path/filepath"
 	"syscall"
 
 	"github.com/marmos91/dittofs/internal/logger"
@@ -274,100 +272,6 @@ func getConfigSource(configFile string) string {
 		return config.GetDefaultConfigPath()
 	}
 	return "defaults"
-}
-
-// startDaemon starts the server as a background daemon process.
-func startDaemon() error {
-	// Determine state directory for PID and log files
-	stateDir := os.Getenv("XDG_STATE_HOME")
-	if stateDir == "" {
-		homeDir, err := os.UserHomeDir()
-		if err != nil {
-			return fmt.Errorf("failed to get home directory: %w", err)
-		}
-		stateDir = filepath.Join(homeDir, ".local", "state")
-	}
-	dittofsStateDir := filepath.Join(stateDir, "dittofs")
-
-	// Create state directory if it doesn't exist
-	if err := os.MkdirAll(dittofsStateDir, 0755); err != nil {
-		return fmt.Errorf("failed to create state directory: %w", err)
-	}
-
-	// Set default PID file if not specified
-	pidPath := pidFile
-	if pidPath == "" {
-		pidPath = filepath.Join(dittofsStateDir, "dittofs.pid")
-	}
-
-	// Check if already running
-	if _, err := os.Stat(pidPath); err == nil {
-		pidData, err := os.ReadFile(pidPath)
-		if err == nil {
-			var pid int
-			if _, err := fmt.Sscanf(string(pidData), "%d", &pid); err == nil {
-				// Check if process is still running
-				if process, err := os.FindProcess(pid); err == nil {
-					if err := process.Signal(syscall.Signal(0)); err == nil {
-						return fmt.Errorf("DittoFS is already running (PID %d)\nUse 'dittofs stop' to stop the running instance", pid)
-					}
-				}
-			}
-		}
-		// Stale PID file, remove it
-		_ = os.Remove(pidPath)
-	}
-
-	// Set default log file if not specified
-	logPath := logFile
-	if logPath == "" {
-		logPath = filepath.Join(dittofsStateDir, "dittofs.log")
-	}
-
-	// Get the executable path
-	executable, err := os.Executable()
-	if err != nil {
-		return fmt.Errorf("failed to get executable path: %w", err)
-	}
-
-	// Build arguments for the daemon process
-	daemonArgs := []string{"start", "--foreground", "--pid-file", pidPath}
-	if GetConfigFile() != "" {
-		daemonArgs = append(daemonArgs, "--config", GetConfigFile())
-	}
-
-	// Create the daemon process
-	cmd := exec.Command(executable, daemonArgs...)
-
-	// Open log file for stdout/stderr
-	logFileHandle, err := os.OpenFile(logPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
-	if err != nil {
-		return fmt.Errorf("failed to open log file: %w", err)
-	}
-
-	cmd.Stdout = logFileHandle
-	cmd.Stderr = logFileHandle
-
-	// Detach from parent process
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		Setsid: true,
-	}
-
-	// Start the daemon
-	if err := cmd.Start(); err != nil {
-		_ = logFileHandle.Close()
-		return fmt.Errorf("failed to start daemon: %w", err)
-	}
-
-	_ = logFileHandle.Close()
-
-	fmt.Printf("DittoFS started in background (PID %d)\n", cmd.Process.Pid)
-	fmt.Printf("  PID file: %s\n", pidPath)
-	fmt.Printf("  Log file: %s\n", logPath)
-	fmt.Println("\nUse 'dittofs stop' to stop the server")
-	fmt.Println("Use 'dittofs status' to check server status")
-
-	return nil
 }
 
 // createAdapterFactory returns a factory function that creates protocol adapters

--- a/cmd/dfs/commands/stop.go
+++ b/cmd/dfs/commands/stop.go
@@ -1,11 +1,11 @@
 package commands
 
 import (
+	"errors"
 	"fmt"
 	"os"
 	"strconv"
 	"strings"
-	"syscall"
 
 	"github.com/spf13/cobra"
 )
@@ -15,29 +15,32 @@ var (
 	stopForce   bool
 )
 
+// errProcessDone is a sentinel returned by stopProcess when the process has already exited.
+var errProcessDone = errors.New("process already done")
+
 var stopCmd = &cobra.Command{
 	Use:   "stop",
 	Short: "Stop the DittoFS server",
 	Long: `Stop a running DittoFS server.
 
-By default, sends SIGTERM for graceful shutdown. Use --force for immediate
-termination with SIGKILL.
+By default, sends a graceful shutdown signal. Use --force for immediate
+termination.
 
 Examples:
   # Stop server (uses default PID file)
-  dittofs stop
+  dfs stop
 
   # Stop server using custom PID file
-  dittofs stop --pid-file /var/run/dittofs.pid
+  dfs stop --pid-file /var/run/dittofs.pid
 
-  # Force stop (SIGKILL)
-  dittofs stop --force`,
+  # Force stop
+  dfs stop --force`,
 	RunE: runStop,
 }
 
 func init() {
 	stopCmd.Flags().StringVar(&stopPidFile, "pid-file", "", "Path to PID file (default: $XDG_STATE_HOME/dittofs/dittofs.pid)")
-	stopCmd.Flags().BoolVarP(&stopForce, "force", "f", false, "Force kill (SIGKILL) instead of graceful shutdown (SIGTERM)")
+	stopCmd.Flags().BoolVarP(&stopForce, "force", "f", false, "Force kill instead of graceful shutdown")
 }
 
 func runStop(cmd *cobra.Command, args []string) error {
@@ -68,25 +71,14 @@ func runStop(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("failed to find process %d: %w", pid, err)
 	}
 
-	// Send signal
-	var sig syscall.Signal
-	if stopForce {
-		sig = syscall.SIGKILL
-		fmt.Printf("Sending SIGKILL to process %d...\n", pid)
-	} else {
-		sig = syscall.SIGTERM
-		fmt.Printf("Sending SIGTERM to process %d...\n", pid)
-	}
-
-	if err := process.Signal(sig); err != nil {
-		// Check if process already exited
-		if err == os.ErrProcessDone {
+	// Send signal (platform-specific)
+	if err := stopProcess(process, pid, stopForce); err != nil {
+		if errors.Is(err, errProcessDone) {
 			fmt.Println("Server already stopped")
-			// Clean up PID file
 			_ = os.Remove(pidPath)
 			return nil
 		}
-		return fmt.Errorf("failed to send signal: %w", err)
+		return err
 	}
 
 	if stopForce {

--- a/cmd/dfs/commands/stop_unix.go
+++ b/cmd/dfs/commands/stop_unix.go
@@ -1,0 +1,29 @@
+//go:build !windows
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+// stopProcess sends the appropriate signal to stop the DittoFS server process.
+func stopProcess(process *os.Process, pid int, force bool) error {
+	sig, name := syscall.SIGTERM, "SIGTERM"
+	if force {
+		sig, name = syscall.SIGKILL, "SIGKILL"
+	}
+
+	fmt.Printf("Sending %s to process %d...\n", name, pid)
+
+	err := process.Signal(sig)
+	if err == os.ErrProcessDone {
+		return errProcessDone
+	}
+	if err != nil {
+		return fmt.Errorf("failed to send signal: %w", err)
+	}
+
+	return nil
+}

--- a/cmd/dfs/commands/stop_windows.go
+++ b/cmd/dfs/commands/stop_windows.go
@@ -1,0 +1,30 @@
+//go:build windows
+
+package commands
+
+import (
+	"fmt"
+	"os"
+)
+
+// stopProcess terminates the DittoFS server process on Windows.
+// Force mode uses process.Kill(); graceful mode sends os.Interrupt.
+func stopProcess(process *os.Process, pid int, force bool) error {
+	var err error
+	if force {
+		fmt.Printf("Killing process %d...\n", pid)
+		err = process.Kill()
+	} else {
+		fmt.Printf("Sending interrupt to process %d...\n", pid)
+		err = process.Signal(os.Interrupt)
+	}
+
+	if err == os.ErrProcessDone {
+		return errProcessDone
+	}
+	if err != nil {
+		return fmt.Errorf("failed to stop process: %w", err)
+	}
+
+	return nil
+}

--- a/pkg/cache/wal/mmap_windows.go
+++ b/pkg/cache/wal/mmap_windows.go
@@ -28,8 +28,13 @@ func (p *MmapPersister) Sync() error {
 	return ErrUnsupportedPlatform
 }
 
+// AppendBlockUploaded is not supported on Windows.
+func (p *MmapPersister) AppendBlockUploaded(_ string, _, _ uint32) error {
+	return ErrUnsupportedPlatform
+}
+
 // Recover is not supported on Windows.
-func (p *MmapPersister) Recover() ([]BlockWriteEntry, error) {
+func (p *MmapPersister) Recover() (*RecoveryResult, error) {
 	return nil, ErrUnsupportedPlatform
 }
 

